x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingContext.java

```diff
@@ -48,48 +48,78 @@ class MlAutoscalingContext {
     final PersistentTasksCustomMetadata persistentTasks;
 
     MlAutoscalingContext() {
-        anomalyDetectionTasks = List.of();
-        snapshotUpgradeTasks = List.of();
-        dataframeAnalyticsTasks = List.of();
-        modelAssignments = Map.of();
-
-        waitingAnomalyJobs = List.of();
-        waitingSnapshotUpgrades = List.of();
-        waitingAnalyticsJobs = List.of();
-        waitingAllocatedModels = List.of();
+        this(List.of(), List.of(), List.of(), Map.of(), List.of(), null);
```

Comment:
all changes in this file are refactorings, required to mock this context object in the added tests